polyply/src/meta_molecule.py

```diff
@@ -62,6 +105,40 @@ def add_monomer(self, current, resname, connections):
     def get_edge_resname(self, edge):
         return [self.nodes[edge[0]]["resname"], self.nodes[edge[1]]["resname"]]
 
+    def split_residue(self, split_string, max_resid):
+        """
+        split_string RESNAME:NEW_RESNAME-ATOMS
+        """
+        resname = split_string.split(":")[0]
+        new_residues = split_string.split(":")[1:]
+
+        self.old_resids = nx.get_node_attributes(self.molecule, "resid")
+        self.old_resnames = nx.get_node_attributes(self.molecule, "resname")
+
+        for node in self.nodes:
+            old_resid = self.nodes[node]["resid"]
+            if self.nodes[node]["resname"] == resname:
+                old_atoms = find_atoms(self.molecule, "resid", old_resid)
+                atom_names = {self.molecule.nodes[atom]["atomname"]:atom for atom in old_atoms}
+                for new_res in new_residues:
+                    new_name, atoms = new_res.split("-")
+                    names = atoms.split(",")
+                    for atom in names:
+                        try:
+                            node_key = atom_names[atom]
+                        except KeyError:
+                            msg = ("Residue {} {} has no atom {}.")
+                            raise IOError(msg.format(resname, old_resid, atom))
```

Comment:
What does this do?

Reply:
From https://docs.python.org/3/library/exceptions.html

> When raising a new exception (rather than using a bare raise to re-raise the exception currently being handled), the implicit exception context can be supplemented with an explicit cause by using from with raise:
> `raise new_exc from original_exc`
> The expression following from must be an exception or None. It will be set as __cause__ on the raised exception. Setting __cause__ also implicitly sets the __suppress_context__ attribute to True, so that using raise new_exc from None effectively replaces the old exception with the new one for display purposes (e.g. converting KeyError to AttributeError), while leaving the old exception available in __context__ for introspection when debugging.
> The default traceback display code shows these chained exceptions in addition to the traceback for the exception itself. An explicitly chained exception in __cause__ is always shown when present. An implicitly chained exception in __context__ is shown only if __cause__ is None and __suppress_context__ is false.